poppy/tests/test_core.py

```diff
@@ -393,3 +393,11 @@ def test_unit_conversions():
         difference = numeric_psf[0].data-analytic
         assert np.all(np.abs(difference) < 3e-5)
 
+def test_return_final_wf():
+    osys =poppy_core.OpticalSystem()
+    osys.add_pupil(optics.CircularAperture(radius=3))   
+    osys.add_detector(pixelscale=0.010, fov_arcsec=5.0)
+    psf = osys.calc_psf(2e-6,return_final=True) 
+    assert len(psf[1])==1 #make sure only one element was returned
+    #test that the wavefront returned is the final wavefront:
+    assert np.max(psf[1][0].intensity-psf[0][0].data)**2 ==0 
```

Comment:
Another way of doing this test that might be a little better would be
```assert np.allclose(psf[1][0].intensity, psf[0][0].data)```

Using `allclose` is more robust since it would ignore any differences around machine precision in the least significant bits. I agree it seems reasonable to expect exact agreement here but I'm not 100% certain that would hold try in all cases. 

I'm also a little confused about the `**2` since that's outside the call to `np.max()`; is that intentional?